src/LiteralInteger.hs

```diff
@@ -0,0 +1,24 @@
+{-# LANGUAGE TemplateHaskell #-}
+module LiteralInteger where
+
+import Control.Lens
+import Control.Monad
+
+import Alpha
+import ShortShow
+
+
+-- | A newtype wrapping user-supplied literal integers.
+--
+-- Useful to have a newtype in case we allow, e.g., integer literals in another
+-- base someday.
```

Comment:
Do you buy this?

Reply:
Would the idea be to replace the newtype with a sum type, with a constructor for each base, so that we can print syntax objects as the user wrote them? Seems fine enough, but also like something we don't need to really plan a lot for.

Reply:
Removed